pkg/query/flamegraph_table.go

```diff
@@ -475,3 +481,85 @@ func equalsByNameTable(tables TableGetter, a, b *querypb.FlamegraphNode) bool {
 	strings := tables.Strings()
 	return strings[aFunction.NameStringIndex] == strings[bFunction.NameStringIndex]
 }
+
+type FlamegraphChildren []*querypb.FlamegraphNode
+
+func (n FlamegraphChildren) Cumulative() int64 {
+	cumulative := int64(0)
+	for _, child := range n {
+		cumulative += child.Cumulative
+	}
+	return cumulative
+}
+
+func (n FlamegraphChildren) Diff() int64 {
+	diff := int64(0)
+	for _, child := range n {
+		diff += child.Diff
+	}
+	return diff
+}
+
+func TrimFlamegraph(ctx context.Context, tracer trace.Tracer, graph *querypb.Flamegraph, thresholdRate float32) *querypb.Flamegraph {
+	ctx, span := tracer.Start(ctx, "trimFlamegraph")
+	defer span.End()
+	if graph == nil {
+		return nil
+	}
+	total := graph.Total
+
+	threshold := int64(float64(thresholdRate) * float64(total))
+	var children FlamegraphChildren = trimFlamegraphNodes(ctx, tracer, graph.Root.Children, threshold)
+	newTotal := int64(0)
+	newDiff := int64(0)
+	if len(graph.Root.Children) > 0 {
+		newTotal = children.Cumulative()
```

Comment:
I think we should include the difference in cumulative due to node trimming in the proto response. I think it would be very confusing to click on a sample on a graph with let's say 100 CPU samples, and then only see 85 cumulative in the root node.

Reply:
In that case, how do we account for the difference in the Child nodes? As the sum of child nodes would still only be 85 right?

Reply:
Sorry I meant ultimately the UI should show something like "Showing 85 out of 100 samples"

Reply:
So actually I meant that we should include the previous and trimmed total value in the response

Reply:
Got it.

Reply:
Is `samples` the right terminology for all profile types? 
Also would it be the right place to show this message or should we hide it under some `i` icon?

<img width="1475" alt="Screenshot 2022-12-21 at 12 01 21 PM" src="https://user-images.githubusercontent.com/1283424/208837261-42c5e5f0-6e8b-4f32-8f25-40cbd9df543f.png">

Reply:
This is done.